src/lang/std/engineConnection.ts

```diff
@@ -557,15 +773,8 @@ class EngineConnection {
     this.pc = undefined
     this.unreliableDataChannel = undefined
     this.webrtcStatsCollector = undefined
-    if (this.failedConnTimeout) {
-      console.log('closed timeout in close')
-      clearTimeout(this.failedConnTimeout)
-      this.failedConnTimeout = null
-    }
 
```

Comment:
Some don't because `this.close()` causes many `close` events to fire, and those subsequently call `this.close()` to guarantee complete "clean up". This can destroy the original purpose of disconnecting, such as by timeout, if we write to `this.state` again after every `this.close()`. I agree it's not ideal and we miss some information here. From what I understand `close` events are non-fatal closings, so keeping the original disconnect reason (quit or timeout) aligns with things, vs `error` events changing the disconnect reason to, well, being an error.

Reply:
I'm open to ideas here. `this.close()` is actually more like `this.cleanup()`... Which helps me accept some invocations don't have `this.state = ...` ...

Reply:
I've got something. :)